File: aat-lib/src/main/java/ch/bailu/aat_lib/description/NameDescription.java
package ch.bailu.aat_lib.description;


import ch.bailu.aat_lib.gpx.GpxInformation;
import ch.bailu.aat_lib.resources.Res;

public class NameDescription extends ContentDescription {
    private String name = "";

    @Override
    public String getLabel() {
        return Res.str().d_name();
    }

    @Override
    public String getUnit() {
        return "";
    }

    public String getValue() {
        return name;
    }

    public boolean updateName(String s) {
        boolean r = !name.equals(s);
        name=s;
        return r;
    }

    @Override
    public void onContentUpdated(int iid, GpxInformation info) {
        updateName(info.getFile().getName());
    }

}

File: aat-lib/src/main/java/ch/bailu/aat_lib/description/NameDescription.kt
package ch.bailu.aat_lib.description

import ch.bailu.aat_lib.gpx.GpxInformation
import ch.bailu.aat_lib.resources.Res

open class NameDescription : ContentDescription() {
    private var name = ""
    override fun getLabel(): String {
        return Res.str().d_name()
    }

    override fun getUnit(): String {
        return ""
    }

    override fun getValue(): String {
        return name
    }

    fun updateName(s: String): Boolean {
        val r = name != s
        name = s
        return r
    }

    override fun onContentUpdated(iid: Int, info: GpxInformation) {
        updateName(info.file.name)
    }
}
